File: solutions/1684/java/Solution.java
// 1684. Count the Number of Consistent Strings
class Solution
{
	public int countConsistentStrings(String allowed, String[] words)
	{
		HashSet<Character> allowedSet = new HashSet<Character>();

		for (int i = 0; i < allowed.length(); ++i)
		{
			allowedSet.add(allowed.charAt(i));
		}

		int allowedCount = words.length;

		for (int i = 0; i < words.length; ++i)
		{
			for (int j = 0; j < words[i].length(); ++j)
			{
				if (!allowedSet.contains(words[i].charAt(j)))
				{
					--allowedCount;

					break;
				}
			}
		}

		return allowedCount;
	}
}

File: solutions/1684/js/solution.js
// 1684. Count the Number of Consistent Strings
var countConsistentStrings = function(allowed, words)
{
	let allowedSet = new Set();

	for (let i = 0; i < allowed.length; ++i)
	{
		allowedSet.add(allowed[i]);
	}

	let allowedCount = words.length;

	for (let i = 0; i < words.length; ++i)
	{
		for (let j = 0; j < words[i].length; ++j)
		{
			if (!allowedSet.has(words[i][j]))
			{
				--allowedCount;

				break;
			}
		}
	}

	return allowedCount;
};